tools/ctl/ide/ui.go

```diff
@@ -207,6 +218,65 @@ func (e *explorer) showDetails(ctx context.Context, example rundex.Rebuild) {
 	showModal(e.app, e.container, details, modalOpts{Margin: 10})
 }
 
+func downloadLogs(ctx context.Context, localAssets rebuild.AssetStore, example rundex.Rebuild) error {
+	t := rebuild.Target{
+		Ecosystem: rebuild.Ecosystem(example.Ecosystem),
+		Package:   example.Package,
+		Version:   example.Version,
+		Artifact:  example.Artifact,
+	}
+	logsAsset := rebuild.Asset{Target: t, Type: rebuild.DebugLogsAsset}
+	debugAssets, err := rebuild.DebugStoreFromContext(context.WithValue(ctx, rebuild.RunID, example.RunID))
+	if err != nil {
+		return errors.Wrap(err, "failed to create debug asset store")
+	}
+	// First, try to get the logs from the debug store directly.
+	if rebuild.AssetCopy(ctx, localAssets, debugAssets, logsAsset) == nil {
+		return nil
+	}
+	// If directly copying from the debug store doesn't work, attempt to find GCB logs.
+	logsBucket, ok := ctx.Value(rebuild.LogsBucketID).(string)
+	if !ok {
+		return errors.New("cannot read logs from gcb, not bucket provided")
+	}
+	var info rebuild.BuildInfo
+	{
+		infoAsset := rebuild.Asset{Target: t, Type: rebuild.BuildInfoAsset}
+		r, err := debugAssets.Reader(ctx, infoAsset)
+		if err != nil {
+			return errors.Wrap(err, "reading build info")
+		}
+		if json.NewDecoder(r).Decode(&info) != nil {
+			return errors.Wrap(err, "parseing build info")
+		}
+	}
+	if info.BuildID == "" {
+		return errors.New("BuildID is empty, cannot read gcb logs")
+	}
+	client, err := gcs.NewClient(ctx)
+	if err != nil {
+		return errors.Wrap(err, "creating gcs client")
+	}
+	obj := client.Bucket(logsBucket).Object(fmt.Sprintf("log-%s.txt", info.BuildID))
```

Comment:
honestly, the log name helper might fit better in the GCB internal library. this isn't obvious and probably deserves some central interface to ensure we get it right (where we could also document/link to the other step-wise log files).

Reply:
Moved to internal lib, added comment about per-step logs.